src/TriMesh.cpp

```diff
@@ -144,9 +144,9 @@ bool TriMesh::collapse_edge(const Tuple& loc0, Tuple& new_t)
     size_t test_fid2 = loc1.get_fid();
 
     assert(
-        ((std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), test_fid1) &&
-          std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), test_fid2)),
-         "faces at the edge is not correct"));
+        std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), test_fid1) > 0 &&
```

Comment:
Might as well write an auxiliary function called `contains` that takes a vector or T, a T, and does a std::find to check if the element is present.

Reply:
Agree,
@yfzhou42 could you fix this too?